Adding Confidential Computing Attestation to Admin Guide (#2950)
uyuni-project/uyuni-docs: modules/administration/pages/confidential-computing.adoc

> [[confidental_computing]]
> = Confidental Computing 

Reviewer: ```suggestion
[[confidential_computing]]
= Confidential Computing 
```

[role=procedure]
. For the selected system, go to tab menu:Audit[Confidential Computing > Settings]
. Enable the attestation by selecting the toggle button. 
. In the field [literal]``Environment Type`` select the correct option from the drop-down list.
//. (PLACEHOLDER) Enable the automatic attestaion at reboot by selecting the toggle button [literal]``Execution``.
. Click button btn:[Save] to save the changes.
+
image::attestation1.jpg[]
//TODO: ALL images will be converted to PNG format when the latest versions are captured

.Procedure: Scheduling New Attestation
[role=procedure]
. For the selected system, go to tab menu:Audit[Confidential Computing > List Attestations]
. Click btn:[Schedule Attestation].
  The new form opens. 
. In the field [literal]``Earliest`` select the time of running the attestation.
. If needed, add the newly created attestation to the action chain by selecting [literal]``Add to`` option.
. Click button btn:[Schedule] to save and schedule the new attestation execution.
+
image::attestation2.jpg[]
+

.Procedure: Viewing Attestation Reports
[role=procedure]
. For the selected system, go to tab menu:Audit[Confidential Computing > List Attestations].

> . From the list of shown, find the report you want to view.

Reviewer: ```suggestion
. Select the report you want to view.
```

+
image::attestation3.jpg[]
+
. In the column [literal]``Actions`` click the icon to open the report.
  Tab [literal]``Overview`` will open.
+
image::attestation4.jpg[]
+
. Move to the next tab [literal]``SEV-SNP``.
// This tab does or does not (NOTE: confirm dependance on the actual environment types) is available depending on the Environment Type selected at the time of scheduling.
+
image::attestation5.jpg[]
+
//. (PLACEHOLDER) To check the details about secure booting, open tab [literal]``Secure Boot``.
//+
//image::attestation6.jpg[]

=== Report statuses

Attestation reports can have one of the following statuses:

Pending::
This is the default status of the scheduled attestation.

> The reposrt is still not available, ether because the process has not yet started or completed.

Reviewer: ```suggestion
The report is still not available, either because the process has not yet started or completed.
```

> Succesful::

Reviewer: ```suggestion
Successful::
```

When the scheduled attestation creates a report which can be viewed, the status of the process is [literal]``Successful``.

Failed::
When the scheduled fails and does not create a report as a result, the status of the process is [literal]``Failed``.

> == Related topics

Reviewer: ```suggestion
== For More Information
```

[[confidential_computing]]
= Confidential Computing 

{CoCo} is a technology which allows protection of data in use by using hardware-based Trusted Execution Environment (TEE), the type of environments that provide increased level of security for data integrity, data confidentiality, and code integrity.

== {CoCo} with {productname}

> The trustworthiness of the TTE is checked with the attestation process.

Reviewer 2: ```suggestion
The trustworthiness of the TEE is checked with the attestation process.
```

{productname} can be used as an attestation server for the systems registered to it.
It generates a report page for the systems which run in this mode.
These systems need to be attested and checked on regular base.
The history of the past checks is also stored and available per request.

{CoCo} Attestation depends on the used hardware and environment where the attested systems are running on.

== Requirements

> {CoCo} can be set up in an the environment with the following characteristics:

Reviewer: ```suggestion
{CoCo} can be set up in an environment with the following characteristics:
```

* Attested system (virtual machine) is *SLES15 SP6* and bootstrapped to {productname}

> * Hardware must have *AMD EPYC Milan CPU* or *AMD EPYC Genoa CPU*

Reviewer 2: There are more requirements:

- The BIOS must be configured to allow CoCo Attestation
- The Host OS and the virtualization software (KVM and libvirt) must support it. (SLES15 SP6 has this as technical preview)

//// 
This was from the testing guidelines, perhaps it is not needed in the documentation?
* [literal]``kernel-default`` must be installed instead of [literal]``kernel-default-base``.
  If necessary, run:
+
----
zypper install -- -kernel-default-base +kernel-default
----
////

=== Limitations

> {CoCo}  Attestation is currently implemented only as a technological preview. 

Reviewer: ```suggestion
{CoCo}  Attestation is currently implemented only as a technology preview. 
```

> //OM: (Do we state this explicitly in the documentation?)

Reviewer 2: We should add a link to the setup procedure for the HostOS as soon as such a documentation exists.
But maybe you can write already a sentence like:

"How to setup and configure CoCo on your Host, please have a look at the OS Vendor documentation"

== Use {CoCo} in {productname}

> .Procedure: Enabling Attestation

Reviewer 2: You need also another procedure: "Enable the attestation container"

As only a few people have the hardware and the need for CoCo, the attestation container is disabled by default.
@lucidd can describe how to enable the container.
This is a kind of "Pre Requirement" for this feature

Reviewer 3: The feature can be enabled during installation of susemanager with with `mgradm install podman`. To enable it either  this bit has to be present in the mgradm.yaml confg
```yaml
coco:
    replicas: 1
```
or alternatively the corresponding command line parameter --coco-replicas 1 needs to be passed.
The current plan for enabling disabling the container after installation is something like this `mgradm scale --coco-replicas 1` 1 for enabled 0 for disabled. In the future that number should allow more then one container replica too.

[role=procedure]

> . For the selected system, go to tab menu:Audit[Confidential Computing > Settings]

Reviewer: ```suggestion
. For the selected system, go to tab menu:Audit[Confidential Computing > Settings].
```

{productname} can be used as an attestation server for the systems registered to it.
It generates a report page for the systems which run in this mode.
These systems need to be attested and checked on regular base.
The history of the past checks is also stored and available per request.

{CoCo} Attestation depends on the used hardware and environment where the attested systems are running on.

[IMPORTANT]
====
{CoCo} Attestation is only available on x86_64 architecture.
====

== Requirements

{CoCo} can be set up in an environment with the following characteristics:

* Attested system (virtual machine) is SLES15 SP6 and bootstrapped to {productname}
* Hardware must have [literal]``AMD EPYC Milan CPU`` or [literal]``AMD EPYC Genoa CPU``
* BIOS must be configured to allow {CoCo} attestation
* Host OS and the virtualization software (KVM and libvirt) must support {CoCo}. 
+
[NOTE]
====
At the moment, SLES15 SP6 has {CoCo} as technical preview.
====

=== Limitations

{CoCo}  Attestation is currently implemented only as a technology preview. 
//OM: (Do we state this explicitly in the documentation?)

[IMPORTANT]
====

> For the exact steps for seting up and configuring {CoCo} on your host, refer to the OS Vendor documentation.

Reviewer: ```suggestion
For the exact steps for setting up and configuring {CoCo} on your host, refer to the OS Vendor documentation.
```
